packages/redirect-parser/src/merge.ts

```diff
@@ -10,14 +8,18 @@ import { splitResults } from './results.js'
 export const mergeRedirects = function ({ fileRedirects, configRedirects }) {
   const results = [...fileRedirects, ...configRedirects]
   const { redirects, errors } = splitResults(results)
-  const mergedRedirects = redirects.filter(isUniqueRedirect)
+  const mergedRedirects = removeDuplicates(redirects)
   return { redirects: mergedRedirects, errors }
 }
 
 // Remove duplicates. This is especially likely considering `fileRedirects`
 // might have been previously merged to `configRedirects`, which happens when
 // `netlifyConfig.redirects` is modified by plugins.
-// The latest duplicate value is the one kept.
-const isUniqueRedirect = function (redirect, index, redirects) {
-  return !redirects.slice(index + 1).some((otherRedirect) => isDeepStrictEqual(redirect, otherRedirect))
+// The latest duplicate value is the one kept, hence why we need to reverse the
+// arrays in order to keep said logic.
+const removeDuplicates = function (redirects) {
+  const strRedirects: [string] = redirects.map((v) => JSON.stringify(v)).reverse()
+  // Set takes care of creating a set of non duplicate values and preserves order
+  const set = new Set(strRedirects)
+  return [...set].map((v) => JSON.parse(v)).reverse()
```

Comment:
I think you could avoid parsing the objects back if you did something like:

```ts
const uniqueRedirects = new Set<string>()
const duplicateIndexes = new Set<number>()

redirects.forEach((v, i) => {
  const str = JSON.stringify(v)

  if (uniqueRedirects.has(str)) {
    duplicateIndexes.add(i)
  } else {
    uniqueRedirects.add(str)
  }
})

const mergedRedirects = redirects.filter((_, i) => !duplicateIndexes.has(i))

return mergedRedirects
```

Not sure how much difference this will make performance-wise, but maybe worth a shot?

Reply:
Ah sorry, only saw this now 🤦 after doing something similar for the haders - https://github.com/netlify/build/pull/4653 - where I was forced to go with a different approach I'm thinking it might make sense to use something similar here. Let me know what you think 👍 But I'll run some tests on Monday and that way we'll have more data to compare.

Reply:
Ran some more tests locally using a version of:
- https://github.com/netlify/build/blob/4f11454785041ddadf7673ab5ac87034a527cafc/packages/headers-parser/src/merge.ts#L24-L49

Which I believe is similar to what you're describing in there ☝️ 

Used a 200k+ redirects file in order to better highlight the differences.

Using the current `JSON.parse()` approach:
```
bash-5.1$ /usr/bin/time -lh node bench-redirects.js
Generated 45001 redirects
	3.56s real		3.98s user		0.34s sys
           677801984  maximum resident set size
                   0  average shared memory size
                   0  average unshared data size
                   0  average unshared stack size
              193561  page reclaims
                   0  page faults
                   0  swaps
                   0  block input operations
                   0  block output operations
                   0  messages sent
                   0  messages received
                   0  signals received
                   0  voluntary context switches
                7448  involuntary context switches
         62445986557  instructions retired
         13670960738  cycles elapsed
           658956032  peak memory footprint
```

With the mentioned approach (not using JSON.parse):
```
Generated 45001 redirects
	3.48s real		3.78s user		0.27s sys
           452116480  maximum resident set size
                   0  average shared memory size
                   0  average unshared data size
                   0  average unshared stack size
              138161  page reclaims
                   0  page faults
                   0  swaps
                   0  block input operations
                   0  block output operations
                   0  messages sent
                   0  messages received
                   0  signals received
                   0  voluntary context switches
                7410  involuntary context switches
         59548212541  instructions retired
         12775129961  cycles elapsed
           431560128  peak memory footprint
```

The key difference is the memory usage impact (~200MB), so I'm going to repurpose that method to use her too.